tla/consistency/SingleNode.tla

```diff
@@ -0,0 +1,122 @@
+---- MODULE SingleNode ----
+\* A lightweight specification to define the externally visible behaviour of CCF
+\* This specification has been inspired by https://github.com/tlaplus/azure-cosmos-tla
+\* Where possible, naming should be consistent with https://microsoft.github.io/CCF/main/index.html
+\* SingleNode considers a single node CCF service, so no view changes
+
+EXTENDS ExternalHistory
+
+\* Upper bound of the number of client events
+\* Note that this abstract specification does not model CCF nodes so there's no
+\* constant for the number of nodes
+CONSTANT HistoryLimit
+
+\* Abstract ledger views that contains only client transactions (no signatures)
+\* Indexed by view, each ledger is the ledger associated with leader of that view 
+\* In practice, the ledger of every CCF node is one of these or a prefix for one of these
+\* This could be switched to a tree which can represent forks more elegantly
+VARIABLES ledgerViews
```

Comment:
It's taken me a few reads to grok the "view" here - sometimes it seems like an abstract, overloaded sense of "view", but this clarifies it's a specific consensus view (aka term). Is that mapping strictly required? In most places it seems this could just be `ledgers` ("some collection of (sequences of (view and tx tuples))"), but I assume that breaks somewhere?

Reply:
It was originally `ledgers`, changed in response to https://github.com/microsoft/CCF/pull/5699#discussion_r1341534825.  I've renamed to `ledgerBranches` to see if that helps. The idea behind variable is that it represents all possible ledgers at a given time (regardless of the number of nodes), instead of a concrete set of ledger